pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains transforms and helpers functions for decomposing arbitrary unitary
+operations into elementary gates.
+"""
+import pennylane as qml
+from pennylane import math
+
+
+def _convert_to_su2(U):
+    r"""Check unitarity of a matrix and convert it to :math:`SU(2)` if possible.
+
+    Args:
+        U (array[complex]): A matrix, presumed to be :math:`2 \times 2` and unitary.
+
+    Returns:
+        array[complex]: A :math:`2 \times 2` matrix in :math:`SU(2)` that is
+        equivalent to U up to a global phase.
+    """
+    # Check unitarity
+    if not math.allclose(math.dot(U, math.T(math.conj(U))), math.eye(2), atol=1e-7):
+        raise ValueError("Operator must be unitary.")
+
+    # Compute the determinant
+    det = U[0, 0] * U[1, 1] - U[0, 1] * U[1, 0]
+
+    # Convert to SU(2) if it's not close to 1
+    if not math.allclose(det, [1.0]):
+        exp_angle = -1j * math.cast_like(math.angle(det), 1j) / 2
+        U = math.cast_like(U, exp_angle) * math.exp(exp_angle)
+
+    return U
+
+
+def zyz_decomposition(U, wire):
```

Comment:
I'm actually not sure about this one, because it's not a transform. If we know that this is where a larger set of decompositions is going to live, then it makes sense to have a separate section (or even page) with all that info. But since I went and made it a public method, it should probably live somewhere, so I just added a small subsection to the quantum functions part of the docs.

Reply:
(BTW using `..automodapi` threw an error; which is why I added it the way I did with `autosummary`.)

Reply:
👍